src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -453,10 +517,58 @@ public object VisitTable(TableExpression tableExpression, SearchOptions context)
                         _cteToLimit.Add(_tableExpressionCounter);
                     }
 
+                    // Update target reference cte dictionary
+                    var curLimitCte = TableExpressionName(_tableExpressionCounter + 1);
+
+                    // Add current cte limit to the dictionary
+                    if (includeExpression.Reversed)
+                    {
+                        AddIncludeLimitCte(includeExpression.SourceResourceType, curLimitCte);
+                    }
+                    else
+                    {
+                        // Not reversed and a specific target type is provided as the 3rd part of include value
+                        if (includeExpression.TargetResourceType != null)
+                        {
+                            AddIncludeLimitCte(includeExpression.TargetResourceType, curLimitCte);
+                        }
+                        else if (includeExpression.ReferenceSearchParameter != null)
+                        {
+                            includeExpression.ReferenceSearchParameter.TargetResourceTypes?.ToList().ForEach(t => AddIncludeLimitCte(t, curLimitCte));
+                        }
+                    }
+
+                    // Handle Multiple Results sets to include from
+                    if (_includeFromCteIds?.Count > 1 && _curFromCteIndex >= 0 && _curFromCteIndex < _includeFromCteIds.Count - 1)
+                    {
+                        StringBuilder.Append($"),{Environment.NewLine}");
```

Comment:
Is there an `AppendLine`?

Reply:
There is. I wanted to be consistent with existing code earlier in that class:
                StringBuilder.AppendDelimited($",{Environment.NewLine}", expression.TableExpressions, (sb, tableExpression) =>